tests/CLI/modules/vs_tests.py

```diff
@@ -132,3 +131,35 @@ def test_create(self, confirm_mock):
                  'networkComponents': [{'maxSpeed': '100'}]},)
         self.assert_called_with('SoftLayer_Virtual_Guest', 'createObject',
                                 args=args)
+
+    @mock.patch('SoftLayer.CLI.formatting.confirm')
+    def test_dns_sync(self, confirm_mock):
+        confirm_mock.return_value = True
+        getReverseDomainRecords = self.set_mock('SoftLayer_Virtual_Guest', 
+                                  'getReverseDomainRecords')
+        getReverseDomainRecords.return_value = [{
+            'networkAddress': '172.16.240.2',
+            'name': '2.240.16.172.in-addr.arpa',
+            'resourceRecords': [{'data': 'test.softlayer.com.',
+                                 'id': 100,
+                                 'host': '2'}],
+            'updateDate': '2013-09-11T14:36:57-07:00',
+            'serial': 1234665663,
+            'id': 123456,
+        }]
+        getResourceRecords = self.set_mock('SoftLayer_Dns_Domain', 
+                                           'getResourceRecords')
+        getResourceRecords.return_value = [
+            {'id': 1, 
+             'ttl': 7200, 
+             'data': '172.16.240.2', 
+             'host': 'test', 
+             'type': 'a'}
+        ]
+        result = self.run_command(['vs', 'dns-sync', '100'])
+        self.assertEqual(result.exit_code, 0)
+        result = self.run_command(['vs', 'dns-sync', '--aaaa-record', '100'])
+        self.assertEqual(result.exit_code, 2)
```

Comment:
I've added checks for the proper API calls. And I do realize that exit_code 2 means there was an exception. I've added checks to make sure its CLIAbort exception instead of something else though.

I have also added some more checks to get the code coverage for this function to 100%.